sql/src/test/java/org/apache/druid/sql/calcite/schema/SegmentMetadataCacheTest.java

```diff
@@ -1447,6 +1460,31 @@ public void testStaleDatasourceRefresh() throws IOException, InterruptedExceptio
     Assert.assertNull(schema.getDatasource("wat"));
   }
 
+  @Test
+  public void testRefreshShouldEmitMetrics() throws InterruptedException
+  {
+    ServiceEmitter mockEmitter = Mockito.mock(ServiceEmitter.class);
+    Map<String, ServiceMetricEvent> emittedEvents = new HashMap<>();
+    Mockito.doCallRealMethod().when(mockEmitter).emit(ArgumentMatchers.any(ServiceEventBuilder.class));
+    Mockito
+        .doAnswer(invocation -> {
+          ServiceMetricEvent e = invocation.getArgument(0);
+          emittedEvents.put(e.getMetric(), e);
+          return null;
+        })
+        .when(mockEmitter).emit(ArgumentMatchers.any(Event.class));
+
+    buildSchemaMarkAndTableLatch(SEGMENT_CACHE_CONFIG_DEFAULT, mockEmitter);
+
+    Assert.assertTrue(emittedEvents.containsKey("init/metadatacache/time"));
+    Assert.assertEquals(emittedEvents.get("segment/metadatacache/refresh/count")
+                                     .getUserDims()
+                                     .get(DruidMetrics.DATASOURCE), "some_datasource");
+    Assert.assertEquals(emittedEvents.get("segment/metadatacache/refresh/time")
```

Comment:
Using `StubServiceEmitter.getMetricValues()` will simplify this code.

Reply:
Instead of verifying the exact value, I am verifying that the metrics are emitted with dataSource dimension.

Reply:
`getMetricValues` can be used for that too because it allows you to pass a set of dimension filters. See some of the other tests where it is being used.